apps/archive/archive.py

```diff
@@ -1026,6 +1026,73 @@ def _send_mark_user_notifications(self, activity_name, msg, resource=None, item=
                           user_list=user_list,
                           extension='markForUser')
 
+    def get_items_chain(self, item):
+        """
+        Get the whole items chain which includes all previous updates,
+        all translations and the original item.
+
+        Result list may contain:
+            - original item
+            - original item's translations
+            - update-1 item
+            - update-1 item's translations
+            - update-N item
+            - update-N item's translations
+            - `item`
+            - `item`'s translations
+        NOTE: The first item in the list is the original item
+
+        :param item: item can be an "initial", "rewrite", or "translation"
+        :type item: dict
+        :return: chain of items
+        :rtype: list
+        """
+
+        def get_item_translated_from(item):
+            while True:
+                try:
+                    item = self.find_one(req={}, _id=item['translated_from'])
+                except Exception:
+                    break
+            return item
+
+        item = get_item_translated_from(item)
+        # add item + translations
+        items_chain = [item]
+        items_chain += self.get_item_translations(item)
+
+        while True:
```

Comment:
as a safety measure I would prefer something like `for i in range(0, 50):` to avoid infinite loops